{{cookiecutter.project_name}}/monitoring/model_monitoring_setup_template.py

```diff
@@ -0,0 +1,278 @@
+# If columns are missing on either side then monitoring will use best-effort heuristics to compute the output metrics.
+baseline_table_name = None
+
+# data_monitoring_dir = "/Users/first.last@company.com" The absolute path to user-configurable directory for storing data monitoring assets.
+# The path must be pointing to an existing directory. If provided, the assets will be created under:
+#         "{{data_monitoring_dir}}/{{table_name}}"
+# Otherwise, the assets will be stored under the default directory:
+#         "/Users/{{user_name}}/databricks_data_monitoring/{{table_name}}"
+# Note that this directory can exist anywhere, including "/Shared/" or other locations outside of the /Users/ 
+# directory; this can be useful when configuring a production monitor intended to be shared within an organization.
+data_monitoring_dir = None
+
+# List of column expressions to slice data with for targeted analysis. The data is grouped by each expression independently,
+# resulting in a separate slice for each predicate and its complements. For example `slicing_exprs=["col_1", "col_2 > 10"]` will generate the
+# following slices: two slices for `col_2 > 10` (True and False), and one slice per unique value in `col1`. For high-cardinality columns, only the
+# top 100 unique values by frequency will generate slices.
+slicing_exprs = None
+
+# A list of custom metrics to compute alongside existing aggregate, derived, and drift metrics.
+custom_metrics = []
+
+# List of Databricks entity names that are associated
+# with this table. Only following entities are supported:
+#    ["models:/registry_model_name"] links a model registry model to the monitored table.
+linked_entities = ["models:/" + model_name]
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC 
+# MAGIC ## 1. Create the monitor
+# MAGIC - You can check **API Walkthrogh - Monitors** section from the [user guide](https://docs.google.com/document/d/1d7BUxbxKDnQfhKZ8e0w99by9JLF4gRmAvDjVxmpE0ZU/edit#heading=h.70vskvghxjux) for more explainations.
+
+# COMMAND ----------
+
+inference_analysis = analysis_type.InferenceLog(
+    timestamp_col=timestamp_col,
+    model_version_col=model_version_col,
+    prediction_col=prediction_col,
+    problem_type=problem_type,
+    label_col=label_col,
+    example_id_col=example_id_col
+)
+
+dm.create_or_update_monitor(
+    table_name=table_name,
+    analysis_type=inference_analysis,
+    granularities=granularities,
+    output_schema_name=output_schema_name,
+    baseline_table_name=baseline_table_name,
+    custom_metrics=custom_metrics,
+    linked_entities=linked_entities,
+    data_monitoring_dir=data_monitoring_dir
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC 
+# MAGIC ### Refresh Metrics
+
+# COMMAND ----------
+
+dm.refresh_metrics(
+ table_name = table_name
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC 
+# MAGIC ## Custom Metrics
```

Comment:
Custom metrics would've been already provided at this point, so it seems late to now show a bunch of documentation around it. Also, related to other bullet, why do we want to inline all of this which is already nicely presented in the user guide? The users will need to have signed up for the preview to try this anyways. Furthermore, I thought we weren't expecting users to dig through this notebook and mainly interact with Stacks.